watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -0,0 +1,568 @@
+        default={"flowsheet_costing_block": m.fs.ro_costing}
+    )
+
+    desal.M1.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.ro_costing}
+    )
+    desal.PXR.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.ro_costing}
+    )
+    desal.P3.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.ro_costing,
+            "costing_method_arguments": {"cost_electricity_flow": False},
+        }
+    )
+
+    # Aggregate unit level costs and calculate overall process costs
+    m.fs.zo_costing.cost_process()
+    m.fs.ro_costing.cost_process()
+
+    # Annual disposal of brine
+    m.fs.brine_disposal_cost = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * (
+                m.fs.zo_costing.waste_disposal_cost
+                * pyunits.convert(
+                    m.fs.brine.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            )
+        ),
+        doc="Cost of disposing of brine waste",
+    )
+
+    m.fs.dye_recovery_revenue = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * m.fs.zo_costing.dye_mass_cost
+            * pyunits.convert(
+                m.fs.dye_retentate.flow_mass_comp[0, "dye"],
+                to_units=pyunits.kg / m.fs.zo_costing.base_period,
+            )
+        ),
+        doc="Savings from dye recovered back to the plant",
+    )
+
+    m.fs.water_recovery_revenue = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * m.fs.zo_costing.recovered_water_cost
+            * pyunits.convert(
+                m.fs.permeate.properties[0].flow_vol,
+                to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+            )
+        ),
+        doc="Savings from water recovered back to the plant",
+    )
+
+    # Combine results from costing packages and calculate overall metrics
+    @m.Expression()
+    def total_capital_cost(b):
+        return (
+            pyunits.convert(
+                m.fs.zo_costing.total_capital_cost, to_units=pyunits.USD_2018
+            )
+            + m.fs.ro_costing.total_investment_cost
+        )
+
+    @m.Expression()
+    def total_operating_cost(b):
+        return (
+            pyunits.convert(
+                m.fs.zo_costing.total_fixed_operating_cost,
+                to_units=pyunits.USD_2018 / pyunits.year,
+            )
+            + pyunits.convert(
+                m.fs.zo_costing.total_variable_operating_cost,
+                to_units=pyunits.USD_2018 / pyunits.year,
+            )
+            + m.fs.ro_costing.total_operating_cost
+        )
+
+    @m.Expression()
+    def total_externalities(b):
+        return pyunits.convert(
+            m.fs.water_recovery_revenue
+            + m.fs.dye_recovery_revenue
+            - m.fs.brine_disposal_cost,
```

Comment:
I guess adding the WWTP opened a small can of worms, but maybe we should account for sludge disposal too.

Reply:
I think that sounds good. I think it's a fair assumption to just use one waste disposal cost. We can always segregate these later if we have reason to do so.